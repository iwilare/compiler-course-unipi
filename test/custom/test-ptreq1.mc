void pptr(int *a) {
    if(a != NULL)
        print(*a);
    else
        print(-1);
}

int main() {
    int *p1 = NULL;
    int *p2 = NULL;
    int a = 10;

    p1 = &a;
    p2 = &a;

    pptr(p1);
    pptr(p2);

    *p1 = 20;

    pptr(p1);
    pptr(p2);

    p1 = NULL;

    pptr(p1);
    pptr(p2);
}
