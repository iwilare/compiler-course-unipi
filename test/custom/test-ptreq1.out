10
10
20
20
-1
20
